src/test/java/uk/gov/pay/directdebit/webhook/gocardless/services/GoCardlessPaymentHandlerTest.java

```diff
@@ -184,9 +197,37 @@ public void handle_doNotProcess_noOrgansiationOnGatewayAccount() {
                 .withAction("paid")
                 .toEntity());
 
-        when(mockedGoCardlessEventService.findPaymentForEvent(goCardlessEvent)).thenReturn(goCardlessPaymentFixture.toEntity());
+        when(mockedPaymentService.findPaymentByProviderId(GOCARDLESS, goCardlessEvent.getLinksPayment().get()))
+                .thenReturn(Optional.of(payment));
 
         goCardlessPaymentHandler.handle(goCardlessEvent);
         verify(mockedGoCardlessEventService, never()).storeEvent(goCardlessEvent);
     }
+
+    @Test
+    public void handle_onCreatePaymentGoCardlessEvent_shouldThrowExceptionWhenEventHasNoLinkedPayment() {
+        GoCardlessEvent goCardlessEvent = spy(goCardlessEventFixture
```

Comment:
The test passes with a `spy` object. Do we need it?

Reply:
works without so we've removed it, good catch.